src/assets/AssetsController.test.ts

```diff
@@ -444,67 +444,54 @@ describe('AssetsController', () => {
   });
 
   it('should reject a valid suggested asset via watchAsset', async () => {
-    await new Promise(async (resolve) => {
-      const { result, suggestedAssetMeta } = await assetsController.watchAsset(
-        {
-          address: '0xe9f786dfdd9ae4d57e830acb52296837765f0e5b',
-          decimals: 18,
-          symbol: 'TKN',
-        },
-        'ERC20',
-      );
-      assetsController.rejectWatchAsset('foo');
-      assetsController.rejectWatchAsset(suggestedAssetMeta.id);
-      assetsController.hub.once(`${suggestedAssetMeta.id}:finished`, () => {
-        expect(assetsController.state.suggestedAssets).toHaveLength(0);
-      });
-      result.catch((error) => {
-        expect(error.message).toContain('User rejected to watch the asset.');
-        resolve('');
-      });
+    const { result, suggestedAssetMeta } = await assetsController.watchAsset(
+      {
+        address: '0xe9f786dfdd9ae4d57e830acb52296837765f0e5b',
+        decimals: 18,
+        symbol: 'TKN',
+      },
+      'ERC20',
+    );
+    assetsController.rejectWatchAsset('foo');
+    assetsController.rejectWatchAsset(suggestedAssetMeta.id);
+    assetsController.hub.once(`${suggestedAssetMeta.id}:finished`, () => {
+      expect(assetsController.state.suggestedAssets).toHaveLength(0);
     });
+    await expect(result).rejects.toThrow('User rejected to watch the asset.');
   });
 
   it('should accept a valid suggested asset via watchAsset', async () => {
-    await new Promise(async (resolve) => {
-      const { result, suggestedAssetMeta } = await assetsController.watchAsset(
-        {
-          address: '0xe9f786dfdd9ae4d57e830acb52296837765f0e5b',
-          decimals: 18,
-          symbol: 'TKN',
-        },
-        'ERC20',
-      );
-      result.then((res) => {
-        expect(assetsController.state.suggestedAssets).toHaveLength(0);
-        expect(res).toBe('0xe9f786dfdd9ae4d57e830acb52296837765f0e5b');
-        resolve('');
-      });
-      await assetsController.acceptWatchAsset(suggestedAssetMeta.id);
+    const { result, suggestedAssetMeta } = await assetsController.watchAsset(
+      {
+        address: '0xe9f786dfdd9ae4d57e830acb52296837765f0e5b',
+        decimals: 18,
+        symbol: 'TKN',
+      },
+      'ERC20',
+    );
+    await assetsController.acceptWatchAsset(suggestedAssetMeta.id);
+    await result.then((res) => {
+      expect(assetsController.state.suggestedAssets).toHaveLength(0);
+      expect(res).toBe('0xe9f786dfdd9ae4d57e830acb52296837765f0e5b');
     });
```

Comment:
Nit: this could be restructured to just use `await`, rather than `await result.then`

```suggestion
    const res = await result;
    expect(assetsController.state.suggestedAssets).toHaveLength(0);
    expect(res).toBe('0xe9f786dfdd9ae4d57e830acb52296837765f0e5b');
```

Reply:
There are a few more examples of this later on too. I figured we'd might as well completely flatten this and get rid of the Promise callbacks altogether since you're getting rid of so many of them in this PR already, but it's fine, this can wait for another time.

Someday I'll add a "prefer async/await" lint rule, and we can migrate them then.